service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -852,6 +855,86 @@ public List<FilterSitesResponse> filterSites(
         .collect(Collectors.toList());
   }
 
+  @Override
+  public List<FilterSiteImageResponse> filterSiteImages(
+      JWTData userData, FilterSiteImageRequest filterSiteImageRequest) {
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+
+    Condition filterCondition =
+        SITE_IMAGES.APPROVAL_STATUS.eq(String.valueOf(ImageApprovalStatus.SUBMITTED));
+
+    if (filterSiteImageRequest.getSiteIds() != null)
+      filterCondition = filterCondition.and(SITES.ID.in(filterSiteImageRequest.getSiteIds()));
+    if (filterSiteImageRequest.getNeighborhoodIds() != null)
+      filterCondition =
+          filterCondition.and(NEIGHBORHOODS.ID.in(filterSiteImageRequest.getNeighborhoodIds()));
+    if (filterSiteImageRequest.getSubmittedStart() != null)
+      filterCondition =
+          filterCondition.and(SITE_IMAGES.UPLOADED_AT.ge(filterSiteImageRequest.getSubmittedStart()));
+    if (filterSiteImageRequest.getSubmittedEnd() != null)
+      filterCondition =
+          filterCondition.and(SITE_IMAGES.UPLOADED_AT.le(filterSiteImageRequest.getSubmittedEnd()));
+
+    Result<
+            Record11<
+                Integer,
+                String,
+                Integer,
+                String,
+                String,
+                String,
+                Timestamp,
+                String,
+                Integer,
+                String,
+                String>>
+        imagesFiltered =
+            db.select(
+                    SITE_IMAGES.ID,
+                    SITE_IMAGES.IMAGE_URL,
+                    SITES.ID,
+                    USERS.FIRST_NAME,
+                    USERS.LAST_NAME,
+                    USERS.EMAIL,
+                    SITE_IMAGES.UPLOADED_AT,
+                    SITE_ENTRIES.COMMON_NAME,
+                    SITES.NEIGHBORHOOD_ID,
+                    SITES.ADDRESS,
+                    SITE_IMAGES.APPROVAL_STATUS)
+                .from(SITE_IMAGES)
+                .leftJoin(SITE_ENTRIES)
+                .on(SITE_IMAGES.SITE_ENTRY_ID.eq(SITE_ENTRIES.ID))
+                .leftJoin(SITES)
+                .on(SITES.ID.eq(SITE_ENTRIES.SITE_ID))
+                .leftJoin(NEIGHBORHOODS)
+                .on(SITES.NEIGHBORHOOD_ID.eq(NEIGHBORHOODS.ID))
+                .leftJoin(USERS)
+                .on(USERS.ID.eq(SITE_IMAGES.UPLOADER_ID))
+                .where(filterCondition)
+                .fetch();
+
+    return imagesFiltered.stream()
+        .map(
+            rec -> {
+              String uploaderName = rec.get(USERS.FIRST_NAME) + ' ' + rec.get(USERS.LAST_NAME);
+
+              String dateSubmitted =
+                  rec.get(SITE_IMAGES.UPLOADED_AT).toLocalDateTime().toLocalDate().toString();
```

Comment:
⛏️ Related to the previous comment - instead of processing the upload time here, we can just directly pass the record's field into the response object, and it'll automatically get formatted in the response.